src/Cli/dotnet/commands/dotnet-restore/RestoreCommandParser.cs

```diff
@@ -2,126 +2,148 @@
-                            .ForwardAs("-property:RestorePackagesWithLockFile=true")),
-                    Create.Option(
+                        LocalizableStrings.CmdUseLockFileOptionDescription)
+                            .ForwardAs("-property:RestorePackagesWithLockFile=true"),
+                    new Option<bool>(
                         "--locked-mode",
-                        LocalizableStrings.CmdLockedModeOptionDescription,
-                        Accept.NoArguments()
-                            .ForwardAs("-property:RestoreLockedMode=true")),
-                    Create.Option(
+                        LocalizableStrings.CmdLockedModeOptionDescription)
+                            .ForwardAs("-property:RestoreLockedMode=true"),
+                    new Option<string>(
                         "--lock-file-path",
-                        LocalizableStrings.CmdLockFilePathOptionDescription,
-                        Accept.ExactlyOneArgument()
-                            .With(name: LocalizableStrings.CmdLockFilePathOption)
-                            .ForwardAsSingle(o => $"-property:NuGetLockFilePath={o.Arguments.Single()}")),
-                    Create.Option(
+                        LocalizableStrings.CmdLockFilePathOptionDescription)
+                    {
+                        Argument = new Argument<string>(LocalizableStrings.CmdLockFilePathOption)
+                    }.ForwardAsSingle(o => $"-property:NuGetLockFilePath={o}"),
+                    new Option<bool>(
                         "--force-evaluate",
-                        LocalizableStrings.CmdReevaluateOptionDescription,
-                        Accept.NoArguments()
-                            .ForwardAs("-property:RestoreForceEvaluate=true")) }).ToArray();
-        }
+                        LocalizableStrings.CmdReevaluateOptionDescription)
+                            .ForwardAs("-property:RestoreForceEvaluate=true") })
+                .ToArray();
 
-        public static Option[] AddImplicitRestoreOptions(
-            IEnumerable<Option> commandOptions)
+        public static Command GetCommand()
         {
-            return AddImplicitRestoreOptions(commandOptions, false, false).ToArray();
+            var command = new Command("restore", LocalizableStrings.AppFullName);
+
+            command.AddArgument(SlnOrProjectArgument);
+
+            foreach (var option in FullRestoreOptions())
+            {
+                command.AddOption(option);
+            }
+
+            command.AddOption(CommonOptions.DiagOption());
+
+            return command;
         }
 
-        private static IEnumerable<Option> AddImplicitRestoreOptions(
-            IEnumerable<Option> commandOptions,
-            bool showHelp,
-            bool useShortOptions)
+        public static void AddImplicitRestoreOptions(Command command, bool showHelp = false, bool useShortOptions = false, bool includeRuntimeOption = true, bool includeNoDependenciesOption = true)
```

Comment:
No, not new, these options used to always be included. We can't do that anymore because some commands (like build for example) already have a --runtime or --no-dependencies option. With the old parser it didn't care and would just overwrite the duplicate option, but system.commandline throws if we add an option twice.